mcs/class/corlib/System.IO/LogcatTextWriter.cs

```diff
@@ -48,13 +48,43 @@ public override void WriteLine ()
 			var o = line.ToString ();
 			line.Clear ();
 
+			Log (o);
+			stdout.WriteLine (o);
+		}
+
+		void Log (string message)
+		{
+			const int buffer_size = 512;
+
 			unsafe {
-				fixed (byte *b_appname = appname)
-				fixed (byte *b_message = Encoding.UTF8.GetBytes(o)) {
-					Log (b_appname, 1 << 5 /* G_LOG_LEVEL_MESSAGE */, b_message);
+				if (Encoding.UTF8.GetByteCount (message) < buffer_size) {
+					try {
+						fixed (char *b_message = message) {
+							byte* buffer = stackalloc byte[buffer_size];
+							int written = Encoding.UTF8.GetBytes (b_message, message.Length, buffer, buffer_size - 1);
+							buffer [written] = (byte)'\0';
+
+							Log (buffer);
+						}
+
+						return;
```

Comment:
Does return within try cause a non trivial "local unwind" or is very efficient? In some systems it is slow, and preferable to "fall out" of the try and then return.